### PR TITLE
#5762 [Manager] Upgrade create-react-app to the latest version

### PR DESCRIPTION
**Pull request by konekoya**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #issue_number)
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/tsconfig.json [hunk 1]
```diff
@@ -21,7 +21,8 @@
     "noImplicitReturns": true,
     "strictNullChecks": true,
     "noImplicitThis": true,
-    "isolatedModules": true
+    "isolatedModules": true,
+    "noFallthroughCasesInSwitch": true
```
Comment (konekoya) [inline ohara-manager/client/tsconfig.json:25]: Added by CRA

File: ohara-manager/package.json [hunk 2]
```diff
@@ -88,7 +88,7 @@
     }
   },
   "lint-staged": {
-    "*.{js,ts,tsx}": "yarn lint:client:fix",
+    "./client/{src,cypress}/**/*.{js,ts,tsx}": "yarn lint:client:fix",
```
Comment (konekoya) [inline ohara-manager/package.json:91]: Fix linter trying to lint `eslintrc.js` issue

File: ohara-manager/client/src/api/volumeApi.ts [hunk 3]
```diff
@@ -51,7 +51,7 @@ export const get = (objectKey: ObjectKey) => {
   });
 };
 
-export const getAll = (queryParams?: object) => {
+export const getAll = (queryParams?: Record<string, unknown>) => {
```
Comment (konekoya) [inline ohara-manager/client/src/api/volumeApi.ts:54]: Should stick to either `any` or `unknown` for consistency.

File: ohara-manager/client/src/components/DevTool/Header/Controller/ControllerLog.js [hunk 4]
```diff
@@ -14,7 +14,6 @@
  * limitations under the License.
  */
 
-import React from 'react';
```
Comment (konekoya) [inline ohara-manager/client/src/components/DevTool/Header/Controller/ControllerLog.js:17]: `React` is no longer needed when using `JSX`, see https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#whats-a-jsx-transform. Please note that TS still needs this as the support is not yet added (should be added in TypeScript v4.1) Meaning, we still need to import React when creating a component with TS (TSX)

cc @eechih

File: ohara-manager/client/.eslintrc.js [hunk 5]
```diff
@@ -73,12 +73,25 @@ module.exports = {
...
+    '@typescript-eslint/restrict-template-expressions': 'off',
+    '@typescript-eslint/no-unsafe-call': 'off',
+    '@typescript-eslint/no-unsafe-assignment': 'off',
+    '@typescript-eslint/no-unsafe-return': 'off',
+    '@typescript-eslint/restrict-plus-operands': 'off',
+    '@typescript-eslint/no-floating-promises': 'off',
+    'import/no-anonymous-default-export': 'off',
+    'no-use-before-define': 'off',
+    '@typescript-eslint/no-implied-eval': 'off',
+    '@typescript-eslint/ban-ts-comment': 'off',
```
Comment (konekoya) [inline ohara-manager/client/.eslintrc.js:87]: TODO: enable rules

File: ohara-manager/client/package.json [hunk 6]
```diff
@@ -67,9 +67,9 @@
     "moment": "2.27.0",
     "mui-rff": "2.5.6",
     "normalizr": "3.6.0",
-    "react": "16.13.1",
+    "react": "17.0.1",
     "react-beforeunload": "2.2.2",
-    "react-dom": "16.13.1",
+    "react-dom": "17.0.1",
```
Comment (konekoya) [inline ohara-manager/client/package.json:72]: Also, bump React as CRA now supports React 17.

**Comment by konekoya** [comment]
retry manager-unit

**Comment by konekoya** [comment]
retry manager

**Comment by konekoya** [comment]
Due to a dependency issue, CI will always fail during `yarn setup`...

File: ohara-manager/client/package.json [hunk 7]
```diff
@@ -36,7 +36,7 @@
   "dependencies": {
     "@cypress/browserify-preprocessor": "3.0.1",
     "@cypress/code-coverage": "3.8.1",
-    "@cypress/instrument-cra": "1.3.1",
+    "@cypress/instrument-cra": "1.4.0",
```
Comment (konekoya) [inline ohara-manager/client/package.json:39]: This should also be bumped along with CRA v4...

**Comment by konekoya** [comment]
retry manager-ut

**Comment by konekoya** [comment]
retry manager-it

**Comment by konekoya** [comment]
retry manager-ut

**Comment by konekoya** [comment]
retry manager-it

**Comment by konekoya** [comment]
retry manager-it

**Comment by konekoya** [comment]
retry manager-ut

**Comment by jackyoh** [comment]
retry manager-ut

**Comment by jackyoh** [comment]
retry

**Comment by jackyoh** [comment]
retry

**Comment by jackyoh** [comment]
retry

**Comment by jackyoh** [comment]
retry

**Comment by jackyoh** [comment]
retry

**Comment by jackyoh** [comment]
retry

**Comment by jackyoh** [comment]
retry

**Comment by jackyoh** [comment]
retry

**Comment by konekoya** [comment]
retry manager-e2e

**Comment by konekoya** [comment]
retry manager-it

**Review by eechih — APPROVED** [review]

**Pull request merged**